reaktive/src/nativeCommonMain/kotlin/com/badoo/reaktive/utils/Lock.kt

```diff
@@ -67,8 +68,13 @@ internal actual class Lock {
         override fun await(timeoutNanos: Long) {
             if (timeoutNanos >= 0L) {
                 memScoped {
-                    val t: timespec = alloc { setNanos(getTimeNanos() + timeoutNanos) }
-                    pthread_cond_timedwait(cond.ptr, lockPtr, t.ptr)
+                    // iOS does not support pthread_condattr_setclock() nor clock_gettime(), can't use monotonic time
```

Comment:
If last solution worked well for linux systems, let's override implementation for iOS only in different PR? I will check how to get current time with `foundation` functions. Create issue please to not forget.

Reply:
Created #154